modules/jfx.incubator.input/src/main/java/jfx/incubator/scene/control/input/KeyBinding.java

```diff
@@ -0,0 +1,712 @@
+
+        if (mac) {
+            if (alt) {
+                alt = false;
+                option = true;
+            }
+            if (shortcut) {
+                meta = false;
+                command = true;
+            }
+        } else {
+            if (ctrl) {
+                shortcut = false;
+            }
+        }
+
+        if (alt) {
+            m.add(KCondition.ALT);
+        }
+
+        if (command) {
+            m.add(KCondition.COMMAND);
+        }
+
+        if (ctrl) {
+            m.add(KCondition.CTRL);
+        }
+
+        if (meta) {
+            m.add(KCondition.META);
+        }
+
+        if (option) {
+            m.add(KCondition.OPTION);
+        }
+
+        if (ev.isShiftDown()) {
+            m.add(KCondition.SHIFT);
+        }
+
+        KeyBinding keyBinding = new KeyBinding(key, m);
+        //System.err.println("kb=" + keyBinding + " ev=" + toString(ev)); // FIX
+        return keyBinding;
+    }
+
+    // FIX remove, debug
+//    private static String toString(KeyEvent ev) {
+//        StringBuilder sb = new StringBuilder("KeyEvent{");
+//        sb.append("type=").append(ev.getEventType());
+//        sb.append(", char=").append(ev.getCharacter());
+//
+//        String ch = ev.getCharacter();
+//        int sz = ch.length();
+//        if (sz > 0) {
+//            sb.append("(");
+//            for (int i = 0; i < ch.length(); i++) {
+//                sb.append(String.format("%02X", (int)ch.charAt(i)));
+//            }
+//            sb.append(")");
+//        }
+//
+//        sb.append(", code=").append(ev.getCode());
+//
+//        if (ev.isShiftDown()) {
+//            sb.append(", shift");
+//        }
+//        if (ev.isControlDown()) {
+//            sb.append(", control");
+//        }
+//        if (ev.isAltDown()) {
+//            sb.append(", alt");
+//        }
+//        if (ev.isMetaDown()) {
+//            sb.append(", meta");
+//        }
+//        if (ev.isShortcutDown()) {
+//            sb.append(", shortcut");
+//        }
+//
+//        return sb.append("}").toString();
+//    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("KeyBinding{key=");
+        sb.append(key);
+        sb.append(", modifiers=");
+        sb.append(modifiers);
+        sb.append("}");
+        return sb.toString();
+    }
+
+    /**
+     * Returns the event type for this key binding.
+     * @return KeyEvent
+     */
+//    @Override
+//    public EventType<KeyEvent> getEventType() {
+//        if (isKeyPressed()) {
+//            return KeyEvent.KEY_PRESSED;
+//        } else if (isKeyReleased()) {
+//            return KeyEvent.KEY_RELEASED;
+//        } else {
+//            return KeyEvent.KEY_TYPED;
+//        }
+//    }
+//
+//    @Override
+//    public boolean isEventAcceptable(KeyEvent ev) {
+//        return equals(KeyBinding.from(ev));
+//    }
+
+    /** Key bindings builder */
```

Comment:
Can you expand this a bit? Also, end with a period. At a minimum, something like this:

```
 * A builder for {@code KeyBinding} objects.
```

but maybe also adding an additional sentence or two along with a pointer to the `KeyBinding.builder` methods and the `build` method of this class?